functions.php

```diff
@@ -1063,7 +1063,12 @@ function display_news_schema( $post ) {
 function display_external_stories_list_item( $story_id, $show_description ) {
 	$story_url = get_post_meta( $story_id, 'externalstory_url', True );
 	$story_text = get_post_meta( $story_id, 'externalstory_text', True );
-	$story_source = get_post_meta( $story_id, 'externalstory_source', True );
+	$source_name = wp_get_post_terms( $story_id, 'sources' )[0]->name;
```

Comment:
If this doesn't return anything, it's potentially going to blow up for trying to access a property on a null objects, and/or an out of range index exception.